### DISCUSSION TITLE
Azure DevOps MultiPlatform pipeline

### DISCUSSION THREAD

**Discussion by niphlod** [danielpalme/ReportGenerator#709] [Q&A] [answered]
Hi,
   I have a pipeline that runs a .NET test suite in two stages, one using windows and the other one using linux.

I thought I could "park" coverage.cobertura.xml files on the first stage, and then on the second one download the reports and generate reports in a single sweep, using something like 

`-reports:/home/vsts/work/_temp/**/coverage.cobertura.xml;/home/vsts/work/1/coverreports/**/*coverage.cobertura.xml`

where `/home/vsts/work/_temp/**/coverage.cobertura.xml` is "the current stage" (the linux one) and `/home/vsts/work/1/coverreports/**/*coverage.cobertura.xml` is the output from the previous one (the windows one).

Issue here is that stage with windows has win paths on cobertura.xml files, so reportgenerator fails to find the files to build the report.

How can I generate a coverage report that builds upon both stages  ?

**Comment by danielpalme** [comment, marked as answer]
Hi @niphlod,

_ReportGenerator_ relies on the path information in the coverage files.
In your case you have to adjust the _Windows_ paths from the first stage so that the files can be found in the _Linux_ stage.

You can do this with a litte script before executing _ReportGenerator_.

Powershell example:
```ps1
(gc "/home/vsts/work/1/coverreports/your/path/coverage.cobertura.xml") | % { $_ -replace "C:\Some\Windows\Path", "/home/some/linux/path" } | Out-File "/home/vsts/work/1/coverreports/your/path/coverage.cobertura.xml" -Encoding UTF8
```
The script is just an example. Please adjust the paths as needed in your scenario.
